src/Sampler.cpp

```diff
@@ -20,11 +20,9 @@ namespace cppkin
 
 #endif
 
-    void Sampler::AdvanceSampler() {
-        currentSample = atomic_fetch_add(&sampler, 1) + 1;
-    }
-    bool Sampler::ShouldSample() {
+    bool Sampler::AdvanceSampler() {
```

Comment:
I think the logic should be separated in order to get rid of root and head below.